### DISCUSSION TITLE
How to transform the timestamps output by the model to seconds?

### DISCUSSION THREAD

**Discussion by My-captain** [modelscope/FunASR#1184] [Q&A] [answered]
How to transform the timestamps output by the model to seconds?

**Comment by My-captain** [comment]
看了源码，请问是否÷1000就可以了？

**Comment by LauraGPT** [comment, marked as answer]
Yes. For the standard FunASR `AutoModel.generate()` result, timestamps are in **milliseconds**, so divide both boundaries by `1000.0`.

For character/word timestamps:

```python
result = model.generate(input="audio.wav")
item = result[0]

timestamps_seconds = [
    [start_ms / 1000.0, end_ms / 1000.0]
    for start_ms, end_ms in item.get("timestamp", [])
]
```

For VAD/speaker sentence segments:

```python
sentences_seconds = [
    {
        **segment,
        "start": segment["start"] / 1000.0,
        "end": segment["end"] / 1000.0,
    }
    for segment in item.get("sentence_info", [])
]
```

For example, `[1230, 2480]` becomes `[1.23, 2.48]` seconds. Keep the original millisecond values if you need integer subtitle or alignment boundaries, and only convert fields whose contract is milliseconds.

Current source references:
- [AutoModel result contract: `[[start_ms, end_ms], ...]`](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/funasr/auto/auto_model.py#L475-L480)
- [README sentence example using `start / 1000`](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/README.md#L64-L72)
- [Timestamp postprocessing stores integer milliseconds](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/funasr/utils/timestamp_tools.py#L118-L134)
